qsr_lib/src/qsrlib/qsrlib.py

```diff
@@ -144,28 +165,46 @@ def request_qsrs(self, request_message, reset=True):
         :param reset: Boolean, if to reset the self.request_message, default=True
         :return: QSRlib_Response_Message
         """
-        self.timestamp_request_received = datetime.now()
-        self.request_message = request_message
-        try:
-            world_qsr_trace = self.__qsrs_active[self.request_message.which_qsr].get(input_data=self.request_message.input_data,
-                                                                                     include_missing_data=self.request_message.include_missing_data,
-                                                                                     timestamp_request_received=self.timestamp_request_received,
-                                                                                     qsrs_for=self.request_message.qsrs_for,
-                                                                                     qsr_relations_and_values=self.request_message.qsr_relations_and_values,
-                                                                                     future=self.request_message.future,
-                                                                                     config=self.request_message.config,
-                                                                                     dynamic_args=self.request_message.dynamic_args)
-        except KeyError:
-            print("ERROR (QSR_Lib.request_qsrs): it seems that the QSR you requested (" + self.request_message.which_qsr + ") is not implemented yet or has not been activated")
-            world_qsr_trace = False
-        if reset:
-            self.timestamp_request_received = None
-            self.request_message = None
+        world_qsr_traces = []
+        timestamp_request_received = datetime.now()
+
+        # Checking if future is True when a list of QSRs is given.
+        # If not, print error as the string results do not support multiple
+        # QSRs at the same time
+        if not request_message.future and (type(request_message.which_qsr) == list or type(request_message.which_qsr) == tuple):
```

Comment:
A better (and more compact way) to write `(type(request_message.which_qsr) == list` is `isinstance(request_message.which_qsr, list)`

Nothing major though... I might change it when I test it.